pkg/plugins/policies/core/rules/rules.go

```diff
@@ -196,7 +239,7 @@ func (ss Subset) Intersect(other Subset) bool {
 		}
 		oTags, ok := otherByKeysOnlyPositive[tag.Key]
 		if !ok {
-			return true
+			continue
```

Comment:
what's the reason for changing the `Intersect` method?

Reply:
should we remove the change from this PR then, since it's a different problem with a different PR https://github.com/kumahq/kuma/pull/12353?